packages/react-dom-bindings/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -2233,57 +2256,105 @@ function escapeJSObjectForInstructionScripts(input: Object): string {
   });
 }
 
+const precedencePlaceholderStart = stringToPrecomputedChunk(
+  '<style data-rprec="',
+);
+const precedencePlaceholderEnd = stringToPrecomputedChunk('"></style>');
+
 export function writeInitialResources(
   destination: Destination,
   resources: Resources,
   responseState: ResponseState,
 ): boolean {
-  const explicitPreloadsTarget = [];
-  const remainingTarget = [];
+  const target = [];
 
-  const {precedences, explicitPreloads, implicitPreloads} = resources;
+  const {
+    fontPreloads,
+    precedences,
+    usedStylePreloads,
+    scripts,
+    usedScriptPreloads,
+    explicitStylePreloads,
+    explicitScriptPreloads,
+  } = resources;
+
+  fontPreloads.forEach(r => {
```

Comment:
These closures aren't too bad but you do have the same exact one declared multiple times so could make it just one reused one.